tests/core/pyspec/eth2spec/test/capella/block_processing/test_process_withdrawals.py

```diff
@@ -67,35 +106,81 @@ def test_success_empty_queue(spec, state):
 
 @with_capella_and_later
 @spec_state_test
-def test_success_one_in_queue(spec, state):
-    prepare_withdrawal_queue(spec, state, 1)
+def test_success_one_full_withdrawal(spec, state):
+    fully_withdrawable_indices, partial_withdrawals_indices = prepare_expected_withdrawals(
+        spec, state, num_full_withdrawals=1)
+    assert len(fully_withdrawable_indices) == 1
+    assert len(partial_withdrawals_indices) == 0
 
     next_slot(spec, state)
     execution_payload = build_empty_execution_payload(spec, state)
 
-    yield from run_withdrawals_processing(spec, state, execution_payload)
+    expected_withdrawals = yield from run_withdrawals_processing(spec, state, execution_payload)
+
+    verify_post_state(state, spec, expected_withdrawals, fully_withdrawable_indices, partial_withdrawals_indices)
 
 
 @with_capella_and_later
 @spec_state_test
-def test_success_max_per_slot_in_queue(spec, state):
-    prepare_withdrawal_queue(spec, state, spec.MAX_WITHDRAWALS_PER_PAYLOAD)
+def test_success_one_partial_withdrawal(spec, state):
+    fully_withdrawable_indices, partial_withdrawals_indices = prepare_expected_withdrawals(
+        spec, state, num_partial_withdrawals=1)
+    assert len(fully_withdrawable_indices) == 0
+    assert len(partial_withdrawals_indices) == 1
+    for index in partial_withdrawals_indices:
+        assert state.balances[index] != spec.MAX_EFFECTIVE_BALANCE
```

Comment:
```suggestion
        assert state.balances[index] > spec.MAX_EFFECTIVE_BALANCE
```

although I feel like this is checked later in `verify_post_state` right?

Reply:
`verify_post_state` only checks that they are reduced to `MAX_EFFECTIVE_BALANCE`, this check is only a sanity check that the helper `prepare_expected_withdrawals` worked I suppose. Changing as per suggestion. cc @hwwhww